test/csuite/random_directio/main.c

```diff
@@ -0,0 +1,1127 @@
+}
+
+/*
+ * sleep_wait --
+ *	Wait for a process up to a number of seconds.
+ */
+static void
+sleep_wait(uint32_t seconds, pid_t pid)
+{
+	pid_t got;
+	int status;
+
+	while (seconds > 0) {
+		if ((got = waitpid(pid, &status, WNOHANG|WUNTRACED)) == pid) {
+			if (WIFEXITED(status))
+				testutil_die(EINVAL,
+				    "Child process %" PRIu64 " exited early"
+				    " with status %d", (uint64_t)pid,
+				    WEXITSTATUS(status));
+			if (WIFSIGNALED(status))
+				testutil_die(EINVAL,
+				    "Child process %" PRIu64 " terminated "
+				    " with signal %d", (uint64_t)pid,
+				    WTERMSIG(status));
+		} else if (got == -1)
+			testutil_die(errno, "waitpid");
+
+		--seconds;
+		sleep(1);
+	}
+}
+
+/*
+ * has_direct_io --
+ *	Check for direct I/O support.
+ */
+static bool
+has_direct_io(void)
+{
+#ifdef O_DIRECT
+	return (true);
+#else
+	return (false);
+#endif
+}
+
+/*
+ * main --
+ *	Top level test.
+ */
+int
+main(int argc, char *argv[])
+{
+	struct sigaction sa;
+	struct stat sb;
+	WT_RAND_STATE rnd;
+	pid_t pid;
+	uint32_t datasize, i, interval, ncycles, nth, timeout;
+	int ch, status;
+	const char *method, *working_dir;
+	char buf[1024];
+	bool populate_only, rand_th, rand_time, schema_test, verify_only;
+
+	(void)testutil_set_progname(argv);
+
+	datasize = DEFAULT_DATA_SIZE;
+	nth = MIN_TH;
+	ncycles = DEFAULT_CYCLES;
+	rand_th = rand_time = true;
+	timeout = MIN_TIME;
+	interval = DEFAULT_INTERVAL;
+	populate_only = schema_test = verify_only = false;
+	working_dir = "WT_TEST.random-directio";
+	method = "none";
+	pid = 0;
+
+	if (!has_direct_io()) {
+		fprintf(stderr, "**** test_random_directio: this system does "
+		    "not support direct I/O.\n**** Skipping test.\n");
+		return (EXIT_SUCCESS);
+	}
+	while ((ch = __wt_getopt(progname, argc, argv,
+	    "dh:i:m:n:pST:t:v")) != EOF)
+		switch (ch) {
+		case 'd':
+			datasize = (uint32_t)atoi(__wt_optarg);
+			if (datasize > LARGE_WRITE_SIZE ||
+			    datasize < MIN_DATA_SIZE) {
+				fprintf(stderr,
+				    "-d value is larger than maximum %"
+				    PRId32 "\n",
+				    LARGE_WRITE_SIZE);
+				exit (EXIT_FAILURE);
+			}
+			break;
+		case 'h':
+			working_dir = __wt_optarg;
+			break;
+		case 'i':
+			interval = (uint32_t)atoi(__wt_optarg);
+			break;
+		case 'm':
+			method = __wt_optarg;
+			if (!WT_STREQ(method, "fsync") &&
+			    !WT_STREQ(method, "dsync") &&
+			    !WT_STREQ(method, "none")) {
+				fprintf(stderr,
+				    "-m option requires fsync|dsync|none\n");
+				exit (EXIT_FAILURE);
+			}
+			break;
+		case 'n':
+			ncycles = (uint32_t)atoi(__wt_optarg);
+			break;
+		case 'p':
+			populate_only = true;
+			break;
+		case 'S':
+			schema_test = true;
+			break;
+		case 'T':
+			rand_th = false;
+			nth = (uint32_t)atoi(__wt_optarg);
+			break;
+		case 't':
+			rand_time = false;
+			timeout = (uint32_t)atoi(__wt_optarg);
+			break;
+		case 'v':
+			verify_only = true;
+			break;
+		default:
+			usage();
+		}
+	argc -= __wt_optind;
+	if (argc != 0)
+		usage();
```

Comment:
I recently added a line to `timestamp_abort` that prints out `CONFIG: <args>` for reproducing. That has been very helpful there. (I'd have sworn I also added it to `random_abort` too but it is not in there and should be too. If you add it here, please add it there too.)

Reply:
I'm resolving this here, and opening a new ticket for some followup work on other tests: https://jira.mongodb.org/browse/WT-4319